src/cpp/security/authentication/PKIDH.cpp

```diff
@@ -1051,6 +1052,36 @@ ValidationResult_t PKIDH::validate_local_identity(
         password = &empty_password;
     }
 
+    std::string key_agreement_algorithm = ECDH_prime256v1;
+    std::string* key_agreement_property =
+            PropertyPolicyHelper::find_property(auth_properties, "preferred_key_agreement");
+    if (nullptr != key_agreement_property)
+    {
+        const std::pair<std::string, std::string> key_agreement_allowed_values[] = {
+            {DH_2048_256, DH_2048_256},
+            {ECDH_prime256v1, ECDH_prime256v1},
+            {"ECDH", ECDH_prime256v1},
+            {"RSA", DH_2048_256}
```

Comment:
IDK whether `RSA` refers to `Diffiel Hellman` but to my understanding this should be
```suggestion
            {"DH", DH_2048_256}
```